Assets/MixedRealityToolkit/_Core/Boundary/MixedRealityBoundarySystem.cs

```diff
@@ -66,35 +69,120 @@ public override void Reset()
             InitializeInternal();
         }
 
+        public bool Contains(Vector3 location)
+        {
+            return Contains(location, Boundary.Type.TrackedArea);
+        }
+
+        /// <inheritdoc/>
+        public bool Contains(Vector3 location, Boundary.Type boundaryType)
+        {
+            if (!EdgeUtilities.IsValidPoint(location))
+            {
+                // Invalid location.
+                return false;
+            }
+
+            if (!FloorHeight.HasValue)
+            {
+                // No floor.
+                return false;
+            }
+
+            if ((FloorHeight.Value > location.y) ||
+                (BoundaryHeight < location.y))
+            {
+                // Location below the floor or above the boundary height.
+                return false;
+            }
+
+            // Boundary coordinates are always "on the floor"
+            Vector2 point = new Vector2(location.x, location.z);
+
+            if (boundaryType == Boundary.Type.PlayArea)
+            {
+                // Check the inscribed rectangle.
+                if (InscribedRectangularBounds != null)
+                {
+                    return InscribedRectangularBounds.IsInsideBoundary(point);
+                }
+            }
+            else if(boundaryType == Boundary.Type.TrackedArea)
+            {
+                // Check the geometry
+                return EdgeUtilities.IsInsideBoundary(GeometryBounds, point);
+            }
+
+            // Not in either boundary type.
+            return false;
+        }
+
         /// <summary>
         /// Retrieves the boundary geometry and creates the boundary and inscribed playspace volumes.
         /// </summary>
         private void CalculateBoundaryBounds()
         {
+            // Reset the bounds
+            GeometryBounds = new Edge[0];
+            FloorHeight = null;
+            InscribedRectangularBounds = null;
+
+            // Boundaries are supported for Room Scale experiences only.
             if (XRDevice.GetTrackingSpaceType() != TrackingSpaceType.RoomScale)
             {
-                // Boundaries are supported for Room Scale experiences only.
                 return;
             }
 
-            OutscribedVolume = new Bounds();
+#if !UNITY_WSA
```

Comment:
Why is this here?  it's not even needed.

Managers should not have platform specific code, that is the devices layer's job